net.odamex.Odamex.yml

```diff
@@ -0,0 +1,76 @@
+# yaml-language-server: $schema=https://raw.githubusercontent.com/flatpak/flatpak-builder/main/data/flatpak-manifest.schema.json
+---
+app-id: net.odamex.Odamex
+runtime: org.freedesktop.Platform
+runtime-version: "24.08"
+sdk: org.freedesktop.Sdk
+command: select-exe
+separate-locales: false
+finish-args:
+  - --share=ipc
+  - --device=all # change this to device=input + device=dri when available more widely
+  - --socket=pulseaudio
+  - --share=network
+  - --socket=wayland
+  - --socket=fallback-x11
+  - --filesystem=host
+  - --env=TIMIDITY_CFG=/app/etc/timidity.cfg
+modules:
+  - name: timidity-instruments
+    buildsystem: simple
+    build-commands:
+      - mkdir -p /app/etc/
+      - install timidity.cfg /app/etc/sdl_timidity.cfg
+      - mkdir -p /app/share/timidity/
+      - cp -r instruments /app/share/timidity/instruments
+    sources:
+      - type: archive
+        url: http://www.libsdl.org/projects/mixer/timidity/timidity.tar.gz
+        sha256: ee7c59bb5d5fae206d53eef0e82b5ed739e7cbf69252a3754024f8f52ebe77aa
+  - name: SDL2-mixer
+    buildsystem: cmake-ninja
+    builddir: true
+    config-opts:
+      - -D SDL2MIXER_VENDORED=ON
+      - -D SDL2MIXER_DEPS_SHARED=OFF
+      - -D SDL2MIXER_SAMPLES=OFF
+      - -D CMAKE_INSTALL_LIBDIR=lib
+    sources:
+      - type: git
+        url: https://github.com/libsdl-org/SDL_mixer.git
+        tag: release-2.8.1
+  # cpptrace dependencies
+  - name: zstd
+    buildsystem: cmake-ninja
+    subdir: build/cmake
+    sources:
+      - type: git
+        url: https://github.com/facebook/zstd.git
+        tag: v1.5.7
+  - name: libdwarf
+    buildsystem: cmake-ninja
+    config-opts:
+      - -D BUILD_DWARFDUMP=OFF
+    sources:
+      - type: git
+        url: https://github.com/jeremy-rifkin/libdwarf-lite.git
+        tag: v0.12.0
+  # Odamex (combined)
+  - name: Odamex
+    buildsystem: simple
+    build-commands:
+      - chmod +x packaging/flatpak/flatpak-buildgen.sh
+      - IS_FLATHUB=1
+      - "./packaging/flatpak/flatpak-buildgen.sh"
```

Comment:
why a script instead of a proper build system? Like `buildsystem: cmake-ninja` ?

Reply:
Someone else started the work on this before me and did it this way. It's worked so far in testing but I can change it if it's preferred.

Reply:
unnecessary obscure build script are not a good idea.

Reply:
I checked and there are some extra steps being done between the configure and build steps. I'm looking at the flatpak builder command reference and I don't see a way to run those commands at the time needed.